core/cmd/shell.go

```diff
@@ -197,8 +197,34 @@ func (n ChainlinkAppFactory) NewApplication(ctx context.Context, cfg chainlink.G
 	}
 
 	ds := sqlutil.WrapDataSource(db, appLggr, sqlutil.TimeoutHook(cfg.Database().DefaultQueryTimeout), sqlutil.MonitorHook(cfg.Database().LogSQL))
-
 	keyStore := keystore.New(ds, utils.GetScryptParams(cfg), appLggr)
+
+	err = keyStoreAuthenticator.authenticate(ctx, keyStore, cfg.Password())
+	if err != nil {
+		return nil, errors.Wrap(err, "error authenticating keystore")
+	}
+
+	err = keyStore.CSA().EnsureKey(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to ensure CSA key")
+	}
+
+	csaKeys, err := keyStore.CSA().GetAll()
+	if err != nil {
+		return nil, err
+	}
+	csaKey := csaKeys[0]
+	csaPubKey := csaKey.PublicKey
+	csaPrivKey := csaKey.Raw().Bytes()
+	csaSigner := func(data []byte) []byte {
+		return ed25519.Sign(csaPrivKey, data)
+	}
+
+	err = initGlobals(cfg.Prometheus(), cfg.Tracing(), cfg.Telemetry(), appLggr, csaPubKey, csaSigner)
+	if err != nil {
+		appLggr.Errorf("Failed to initialize globals: %v", err)
```

Comment:
This is pre-existing, but it seems wrong that we just log and carry on in this case :thinking:

Reply:
cc @patrickhuie19 Do you remember this?

Reply:
yea, previously the func didn't return errors. initializing otel tracing (pre-Beholder) was the first reason to return an err. We didn't want to break nodes if the tracing could not start up correctly, and chose to persist this error here.

Reply:
> We didn't want to break nodes if the tracing could not start up correctly,

Does that logic still hold, given beholder requirements?

Reply:
yea, i believe so. My thinking: we don't want an incorrect beholder config to start a node from booting up, especially if there are network issues on app start that could be resolved in the background.

Reply:
I was thinking that errors are no longer permissible because we have an expectation of reporting data reliably via beholder, and so misconfiguration should block startup.

> especially if there are network issues on app start that could be resolved in the background.

What do you mean? I agree network issues should be resolved in the backround. Is that not currently the case? That behavior should be orthogonal to the error returned here though.

Reply:
> That behavior should be orthogonal

Should it? IFIUC, we perform validation of the user supplied Telemetry TOML config before this point in the node startup flow. What class of errors should make it back to core that we want to block startup on?

Reply:
It doesn't have to change in this PR, but we might want to track it somewhere 
cc @mathewdgardner